test/test_charm.py

```diff
@@ -34,13 +35,11 @@ class TestCharmEvents(CharmEvents):
         # We use a subclass temporarily to prevent these side effects from leaking.
         CharmBase.on = TestCharmEvents()
 
-    def tearDown(self):
-        shutil.rmtree(self.tmpdir)
-        os.environ['PATH'] = self._path
-
-        del os.environ['JUJU_UNIT_NAME']
-
-        CharmBase.on = CharmEvents()
+        def cleanup():
+            os.environ['PATH'] = self._path
+            del os.environ['JUJU_UNIT_NAME']
+            CharmBase.on = CharmEvents()
+        self.addCleanup(cleanup)
```

Comment:
We might be able to do this a bit more cleanly if we use [`unittest.mock.patch.dict`](https://docs.python.org/3/library/unittest.mock.html#patch-dict).